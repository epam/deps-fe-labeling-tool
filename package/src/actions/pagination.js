import { createAction } from 'redux-actions'

const FEATURE_NAME = 'PAGINATION'

const openPage = createAction(`${FEATURE_NAME}/OPEN_PAGE`)

const resetDefault = createAction(
  `${FEATURE_NAME}/RESET_DEFAULT`
)

export {
  resetDefault,
  openPage
}
