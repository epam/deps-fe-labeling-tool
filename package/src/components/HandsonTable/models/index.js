
export * from './HTTableData'
export * from './HTMerge'
export * from './HTColumn'
