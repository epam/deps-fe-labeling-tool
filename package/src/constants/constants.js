const CONFIDENCE_ON_MANUAL_CHANGE = 1

export {
  CONFIDENCE_ON_MANUAL_CHANGE
}
