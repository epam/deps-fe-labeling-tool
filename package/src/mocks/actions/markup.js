const addLabels = jest.fn((labels) => ({
  type: 'MOCK_ADD_LABELS_ACTION',
  payload: labels
}))

const addTables = jest.fn((tables) => ({
  type: 'MOCK_ADD_TABLES_ACTION',
  payload: tables
}))

const removeLabels = jest.fn((labels) => ({
  type: 'MOCK_REMOVE_LABELS_ACTION',
  payload: labels
}))

const updateLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_LABELS_ACTION',
  payload: labels
}))

const updateAllLabels = jest.fn((labels) => ({
  type: 'MOCK_UPDATE_ALL_LABELS_ACTION',
  payload: labels
}))

const updateTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_TABLES_ACTION',
  payload: tables
}))

const updateAllTables = jest.fn((tables) => ({
  type: 'MOCK_UPDATE_ALL_TABLES_ACTION',
  payload: tables
}))

const selectLabels = jest.fn((labels) => ({
  type: 'MOCK_SELECT_LABELS_ACTION',
  payload: labels
}))

const selectTables = jest.fn((tables) => ({
  type: 'MOCK_SELECT_TABLES_ACTION',
  payload: tables
}))

const clearSelection = jest.fn(() => ({
  type: 'MOCK_CLEAR_SELECTION_ACTION'
}))

const removeTables = jest.fn((tables) => ({
  type: 'MOCK_REMOVE_TABLES_ACTION',
  payload: tables
}))

const importMarkup = jest.fn(() => ({
  type: 'MOCK_IMPORT_MARKUP'
}))

const exportMarkup = jest.fn(() => ({
  type: 'MOCK_EXPORT_MARKUP'
}))

const undo = jest.fn(() => ({
  type: 'MOCK_UNDO_ACTION'
}))

const redo = jest.fn(() => ({
  type: 'MOCK_REDO_ACTION'
}))

const updateInitialMarkup = jest.fn(() => ({
  type: 'MOCK_UPDATE_INITIAL_MARKUP_ACTION'
}))

const storeMarkup = jest.fn((markup) => ({
  type: 'MOCK_STORE_MARKUP_ACTION',
  payload: markup
}))

const storeAssignedMarkup = jest.fn((markup) => ({
  type: 'MOCK_UPDATE_MARKUP_ACTION',
  payload: markup
}))

const updatePageMarkup = jest.fn(() => ({
  type: 'UPDATE_PAGE_MARKUP'
}))

const reset = jest.fn(() => ({
  type: 'RESET_MARKUP_ACTION'
}))

const copyMarkup = jest.fn(() => ({
  type: 'COPY_MARKUP_ACTION'
}))

const pasteMarkup = jest.fn(() => ({
  type: 'PASTE_MARKUP_ACTION'
}))

const updateLabelsWithSettings = jest.fn(() => ({
  type: 'UPDATE_LABELS_WITH_SETTINGS'
}))

const updateTablesWithSettings = jest.fn(() => ({
  type: 'UPDATE_TABLES_WITH_SETTINGS'
}))

const mockMarkupActions = {
  addLabels,
  addTables,
  copyMarkup,
  pasteMarkup,
  removeLabels,
  reset,
  updateLabels,
  updateAllLabels,
  updateTables,
  updateAllTables,
  selectLabels,
  selectTables,
  clearSelection,
  removeTables,
  importMarkup,
  exportMarkup,
  undo,
  updateInitialMarkup,
  storeMarkup,
  storeAssignedMarkup,
  updatePageMarkup,
  redo,
  updateLabelsWithSettings,
  updateTablesWithSettings
}

export {
  mockMarkupActions
}
