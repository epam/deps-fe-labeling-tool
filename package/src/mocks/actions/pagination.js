const openPage = jest.fn((page) => ({
  type: 'MOCK_OPEN_PAGE_ACTION',
  payload: page
}))

const mockPaginationActions = {
  openPage
}

export {
  mockPaginationActions
}
