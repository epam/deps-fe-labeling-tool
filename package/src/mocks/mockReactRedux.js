const mockDispatch = jest.fn((action) => action)

const createMockMapStateToProps = (mapStateToProps) => {
  if (!mapStateToProps) {
    return mapStateToProps
  }

  return (state, ownProps) => ({
    props: mapStateToProps(state, ownProps)
  })
}

const getDispatchFunctionFormShorthandObject = (actionsObj) => (dispatch) => (
  Object.keys(actionsObj).reduce(
    (props, actionKey) => ({
      ...props,
      [actionKey]: jest.fn(
        (...args) => dispatch(actionsObj[actionKey].apply(undefined, args))
      )
    }),
    {}
  )
)

const createMockMapDispatchToProps = (mapDispatchToProps) => {
  if (!mapDispatchToProps) {
    return mapDispatchToProps
  }

  const mappingFunction = typeof mapDispatchToProps === 'function'
    ? mapDispatchToProps
    : getDispatchFunctionFormShorthandObject(mapDispatchToProps)

  return (dispatch, ownProps) => {
    dispatch = dispatch || mockDispatch

    return {
      props: mappingFunction(dispatch, ownProps),
      dispatch
    }
  }
}

const createMockMergeProps = (mergeProps) => {
  if (!mergeProps) {
    return mergeProps
  }

  return (stateProps, dispatchProps, ownProps) => {
    dispatchProps = dispatchProps || { dispatch: mockDispatch }

    return {
      props: mergeProps(stateProps, dispatchProps, ownProps),
      dispatch: dispatchProps.dispatch
    }
  }
}

const mockReactRedux = {
  connect: (mapStateToProps, mapDispatchToProps, mergeProps) => (WrappedComponent) => ({
    mapStateToProps: createMockMapStateToProps(mapStateToProps),
    mapDispatchToProps: createMockMapDispatchToProps(mapDispatchToProps),
    mergeProps: createMockMergeProps(mergeProps),
    WrappedComponent
  }),
  batch: (fn) => Promise.resolve(fn),
  useDispatch: jest.fn(() => mockDispatch),
  useSelector: jest.fn((selector) => selector())
}

export {
  mockReactRedux
}
