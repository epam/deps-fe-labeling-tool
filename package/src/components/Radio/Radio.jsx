
import React, { PureComponent } from 'react'
import { Radio as AntdRadio } from 'antd'
import 'antd/lib/radio/style/index.less'
import PropTypes from 'prop-types'
import { optionShape } from '@/models/Option'

class RadioGroup extends PureComponent {
  onChange = (e) => {
    this.props.onChange?.(e.target.value)
  }

  render = () => {
    const { options, ...rest } = this.props
    return (
      <AntdRadio.Group
        {...rest}
        onChange={this.onChange}
      >
        {
          options.map(
            (o, i) => (
              <Radio
                key={i}
                value={o.value}
              >
                {o.text}
              </Radio>
            )
          )
        }
      </AntdRadio.Group>
    )
  }
}

RadioGroup.propTypes = {
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(optionShape).isRequired
}

const Radio = (props) => <AntdRadio {...props} />

export {
  Radio,
  RadioGroup
}
