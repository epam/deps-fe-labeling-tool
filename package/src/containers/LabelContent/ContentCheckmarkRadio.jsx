import React, { useCallback } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { updateLabels } from '@/actions/markup'
import { RadioGroup } from '@/components/Radio'
import { CONFIDENCE_ON_MANUAL_CHANGE } from '@/constants/constants'
import { labelShape } from '@/models/Label'
import { Option } from '@/models/Option'
import { currentPageSelector } from '@/selectors/pagination'
import { RadioGroupWrapper } from './ContentCheckmarkRadio.styles'

const RESOURCE_TRUE_OPTION = 'True'
const RESOURCE_FALSE_OPTION = 'False'

const CheckmarkOption = {
  [RESOURCE_FALSE_OPTION]: false,
  [RESOURCE_TRUE_OPTION]: true
}

const options = Object.entries(CheckmarkOption).map(([text, value]) => new Option(value, text))

const ContentCheckmarkRadio = ({
  currentPage,
  label,
  updateLabels
}) => {
  const onContentChange = useCallback(
    (value) => {
      updateLabels(currentPage, [{
        ...label,
        content: value,
        confidence: CONFIDENCE_ON_MANUAL_CHANGE
      }])
    },
    [
      currentPage,
      label,
      updateLabels
    ])

  return (
    <RadioGroupWrapper>
      <RadioGroup
        options={options}
        onChange={onContentChange}
        value={label.content}
      />
    </RadioGroupWrapper>
  )
}

ContentCheckmarkRadio.propTypes = {
  currentPage: PropTypes.number.isRequired,
  label: labelShape.isRequired,
  updateLabels: PropTypes.func.isRequired
}

const mapStateToProps = (state) => ({
  currentPage: currentPageSelector(state)
})

const mapDispatchToProps = {
  updateLabels
}

const ConnectedComponent = connect(mapStateToProps, mapDispatchToProps)(ContentCheckmarkRadio)

export {
  ConnectedComponent as ContentCheckmarkRadio
}
