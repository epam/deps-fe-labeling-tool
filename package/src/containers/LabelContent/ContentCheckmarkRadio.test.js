import { mockMarkupActions } from '@/mocks/actions/markup'
import { mockReactRedux } from '@/mocks/mockReactRedux'
import { mockPaginationSelectors } from '@/mocks/selectors/pagination'
import React from 'react'
import { shallow } from 'enzyme'
import { updateLabels } from '@/actions/markup'
import { RadioGroup } from '@/components/Radio'
import { CONFIDENCE_ON_MANUAL_CHANGE } from '@/constants/constants'
import { Label, LabelType } from '@/models/Label'
import { ContentCheckmarkRadio } from './ContentCheckmarkRadio'

jest.mock('react-redux', () => mockReactRedux)

jest.mock('@/actions/markup', () => mockMarkupActions)
jest.mock('@/selectors/pagination', () => mockPaginationSelectors)

const {
  WrappedComponent,
  mapStateToProps,
  mapDispatchToProps
} = ContentCheckmarkRadio

describe('Container: ContentCheckmarkRadio', () => {
  describe('mapStateToProps', () => {
    it('should pass expected state as props to WrappedComponent', () => {
      const { props } = mapStateToProps()
      const expectedState = {
        currentPage: mockPaginationSelectors.currentPageSelector()
      }
      expect(props).toEqual(expectedState)
    })
  })

  describe('mapDispatchToProps', () => {
    const { props } = mapDispatchToProps()

    it('should pass updateLabels action as prop to WrappedComponent', () => {
      props.updateLabels()
      expect(updateLabels).toHaveBeenCalled()
    })
  })

  describe('component', () => {
    let wrapper, defaultProps

    beforeEach(() => {
      defaultProps = {
        ...mapStateToProps().props,
        ...mapDispatchToProps().props,
        label: new Label(
          0.113,
          0.8104,
          0.017,
          0.0119,
          'enclosures',
          undefined,
          LabelType.CHECKMARK,
          true,
          { data: 'some meta for enclosures value label' },
          0.66
        )
      }
      wrapper = shallow(<WrappedComponent {...defaultProps} />)
    })

    it('should render correct layout based on props', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call props updateLabels with correct argument in case of calling to RadioGroup onChange', () => {
      const radioGroupProps = wrapper.find(RadioGroup).props()
      radioGroupProps.onChange(false)
      expect(defaultProps.updateLabels).nthCalledWith(
        1,
        defaultProps.currentPage,
        [
          {
            ...defaultProps.label,
            content: false,
            confidence: CONFIDENCE_ON_MANUAL_CHANGE
          }
        ])
    })
  })
})
