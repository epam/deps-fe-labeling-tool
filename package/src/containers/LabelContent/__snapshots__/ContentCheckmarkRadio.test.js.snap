// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: ContentCheckmarkRadio component should render correct layout based on props 1`] = `
<ContentCheckmarkRadiostyles__RadioGroupWrapper>
  <RadioGroup
    onChange={[Function]}
    options={
      Array [
        Option {
          "disabled": undefined,
          "text": "False",
          "value": false,
        },
        Option {
          "disabled": undefined,
          "text": "True",
          "value": true,
        },
      ]
    }
    value={true}
  />
</ContentCheckmarkRadiostyles__RadioGroupWrapper>
`;
